src/odemis/driver/simulated.py

```diff
@@ -114,15 +114,15 @@ def _updatePosition(self):
     def _doMoveRel(self, shift):
         maxtime = 0
         for axis, change in shift.items():
-            self._position[axis] += change
             rng = self.axes[axis].range
             if axis in self._inverted:
                 rng = (-rng[1], -rng[0])  # user -> internal range
-            if not rng[0] < self._position[axis] < rng[1]:
-                logging.warning("moving axis %s to %f, outside of range %r",
-                                axis, self._position[axis], rng)
-            else:
-                logging.info("moving axis %s to %f", axis, self._position[axis])
+            if not rng[0] <= self._position[axis] + change <= rng[1]:
+                raise ValueError("moving axis %s to %f, outside of range %r" %
+                                 (axis, self._position[axis] + change, rng))
+
+            self._position[axis] += change
+            logging.info("moving axis %s to %f", axis, self._position[axis])
```

Comment:
It could be nice to add the source of the error in the logging message (simulated stage).
But that is not essential to me.

Reply:
The logging infrastructure already automatically include this info:
`2021-01-23 23:41:52,045 INFO    simulated:125:  moving axis z to 0.000100`